models/Node.js

```diff
@@ -43,8 +43,8 @@ function Node(name, type, objectId, frameId, nodeId) {
         this.publicData = {};
     } else {
         let nodeTemplate = nodeTypes[type];
-        this.privateData = nodeTemplate.properties.privateData || {};
-        this.publicData = nodeTemplate.properties.publicData || {};
+        this.privateData = Object.assign({}, nodeTemplate.properties.privateData) || {};
```

Comment:
publicData's members aren't guaranteed to be primitives as far as I know so this should use a deep copy like `JSON.parse(JSON.stringify(privateData))` instead (it might be nice to introduce a utilities.deepCopy which does a recursive Object.assign but that feels like a bit of a waste of time)

Reply:
Thanks, good catch